tests/integration/solidity_contracts/PlainOpcodes/PlainOpcodes.sol

```diff
@@ -127,4 +128,82 @@ contract PlainOpcodes {
             extcodecopy(target, add(extcode, 0x20), offset, size)
         }
     }
+
+    function testShouldAlwaysRevert() public pure {
+        revert("This always reverts");
+    }
+
+    function testCallingContextShouldPropogateRevertFromSubContext() public {
+        Parent parent = new Parent();
+        parent.triggerRevert();
+
+    }
+
+    function testCallingContextShouldPropogateRevertFromSubContextOnCreate() public {
+        ContractRevertsOnConstruction doomedContract = new ContractRevertsOnConstruction();
+
+    }    
+    
+    function testShouldRevertViaCall() external returns (address address_, bool triggerRevert ) {
+        Parent parent = new Parent();
+
+        (bool success, ) = address(parent).call(abi.encodeWithSignature("triggerRevert()"));
+
+        return (address(parent.child()), success);
+    }
+
+    function testCallingContextShouldCatchRevertFromSubContext() public {
+        Parent parent = new Parent();
+        
+        try parent.triggerRevert() { 
+            // The `triggerRevert` function is expected to revert. 
+            revert("always revert"); 
+        } catch { 
+            // Ensure that the created child contract in parent is deleted. 
+            //  
+            require(address(parent.child()) == address(0), "Parent is deleted"); 
+        } 
+    }   
+
+    function testCallingContextShouldCatchRevertFromSubContextAndPropogateRevertMessage() public {
+        Parent parent = new Parent();
+        
+        try parent.triggerRevert() { 
+            // The `triggerRevert` function is expected to revert. 
+            revert("always revert"); 
+        } catch Error(string memory errorMessage) { 
+            // Ensure that the created child contract in parent is deleted. 
+            require(address(parent.child()) == address(0), "Parent is deleted"); 
+        
+            // Revert with the error message from the subcontext.
+            revert(errorMessage);
+        } 
+    }    
+
+    function testCallingContextShouldCatchRevertFromSubContextWithReversedWrites() public {
```

Comment:
I don't understand the purpose of this test

Reply:
I was basically stress testing the reverting writes logic--that it is restoring the value to the pre-context state. You feel its overkill?